batch/batch/server/database.py

```diff
@@ -1,159 +1,133 @@
-        assert select_fields is None or len(select_fields) != 0
-        async with self._db.pool.acquire() as conn:
-            async with conn.cursor() as cursor:
-                where_template, where_values = make_where_statement(where_items)
-                select_fields = ",".join(select_fields) if select_fields is not None else "*"
-                sql = f"SELECT {select_fields} FROM `{self.name}` WHERE {where_template}"
-                await cursor.execute(sql, where_values)
-                result = await cursor.fetchall()
-        return result
-
-    async def has_record(self, where_items):
-        async with self._db.pool.acquire() as conn:
-            async with conn.cursor() as cursor:
-                where_template, where_values = make_where_statement(where_items)
-                sql = f"SELECT COUNT(1) FROM `{self.name}` WHERE {where_template}"
-                count = await cursor.execute(sql, tuple(where_values))
-        return count >= 1
-
-    async def delete_record(self, where_items):
-        async with self._db.pool.acquire() as conn:
-            async with conn.cursor() as cursor:
-                where_template, where_values = make_where_statement(where_items)
-                sql = f"DELETE FROM `{self.name}` WHERE {where_template}"
-                await cursor.execute(sql, where_values)
+        await super().__init__(config_file)
+
+        self.jobs = await JobsTable(self, os.environ.get('JOBS_TABLE', 'jobs'))
+        self.jobs_parents = await JobsParentsTable(self, os.environ.get('JOBS_PARENTS_TABLE', 'jobs-parents'))
+        self.batch = await BatchTable(self, os.environ.get('BATCH_TABLE', 'batch'))
+        self.batch_jobs = await BatchJobsTable(self, os.environ.get('BATCH_JOBS_TABLE', 'batch-jobs'))
+
+
+class JobsTable(Table):
+    async def __init__(self, db, name='jobs'):
+        schema = {'id': 'BIGINT NOT NULL AUTO_INCREMENT',
+                  'state': 'VARCHAR(40) NOT NULL',
+                  'exit_code': 'INT',
+                  'batch_id': 'BIGINT',
+                  'scratch_folder': 'VARCHAR(1000)',
+                  'pod_name': 'VARCHAR(1000)',
+                  'pvc': 'TEXT(65535)',
+                  'callback': 'TEXT(65535)',
+                  'task_idx': 'INT NOT NULL',
+                  'always_run': 'BOOLEAN',
+                  'time_created': 'TIMESTAMP DEFAULT CURRENT_TIMESTAMP',
+                  'time_ended': 'TIMESTAMP',
+                  'user': 'VARCHAR(1000)',
+                  'attributes': 'TEXT(65535)',
+                  'tasks': 'TEXT(65535)',
+                  'parent_ids': 'TEXT(65535)',
+                  'input_log_uri': 'VARCHAR(1000)',
+                  'main_log_uri': 'VARCHAR(1000)',
+                  'output_log_uri': 'VARCHAR(1000)'}
+
+        keys = ['id']
+
+        await super().__init__(db, name, schema, keys)
+
+    async def update_record(self, id, **items):
+        await super(JobsTable, self).update_record({'id': id}, items)
```

Comment:
Use `super()` throughout.

Reply:
The places where super() was not used was intentional.

Reply:
What's the difference? Isn't super(JobsTable, self) equivalent? https://www.pythonforbeginners.com/super/working-python-super-function

Reply:
Oh sorry. DIdn't understand your comment.

Reply:
done